### PR TITLE
Bid functionality

### PR DESCRIPTION
**Pull request by dangerousScript**

### CODE REVIEW

File: auction9-backend/handler.js [hunk 1]
```diff
@@ -250,3 +252,46 @@ export const realizeFinishedAuction = async (event, context) => {
...
+      let currentHours = todayDate.getHours();
+      let currentMinutes = todayDate.getMinutes();
+      let currentSeconds = todayDate.getSeconds();
+      let formattedTodayDate = currentYear + '-' + currentMonth + '-' + currentDay + ' ' + currentHours + ':' + currentMinutes + ':' + currentSeconds;
+      // first create bid -> update current price with new bid
+      // current userid hardcoded
+      await mysql.query('INSERT INTO tbl_user_auction (`userID`, `auctionID`, `price`, `time`) VALUES (?, ?, ?, ?)', [2, reqBody.auction.auctionID, newBid, formattedTodayDate]);
+      await mysql.end();
+      await mysql.query('UPDATE tbl_auction SET price=? WHERE auctionID=?', [newBid, reqBody.auction.auctionID]);
+      await mysql.end();
```
Comment (mjovanovic) [inline auction9-backend/handler.js:282]: `mysql.end()` should be called only once in the end

File: auction9-backend/serverless.yml [hunk 2]
```diff
@@ -110,3 +110,12 @@ functions:
           path: myauctions/{id}
           method: put
           cors: true
+
+  # creates new bid
+  postNewBid:
+    handler: handler.postNewBid
+    events:
+      - http:
+          path: auctions/{id}/new
```
Comment (mjovanovic) [inline auction9-backend/serverless.yml:119]: REST API URLs should reflect resources being operated with. In that sense better URL would be `auctions/{id}/bids`

File: auction9-backend/handler.js [hunk 3]
```diff
@@ -250,3 +252,46 @@ export const realizeFinishedAuction = async (event, context) => {
...
 };
+
+/* postNewBid - creates new auction
+ * POST: /auctions/id/new
+ */
+export const postNewBid = async (event, context) => {
+  try {
+    let reqBody = JSON.parse(event.body);
+    let newBid = reqBody.newBid;
+    let currentAuctionPrice = await mysql.query('SELECT price FROM tbl_auction WHERE auctionID=?', [reqBody.auction.auctionID]);
```
Comment (mjovanovic) [inline auction9-backend/handler.js:263]: `auctionId` should be used from path parameters

File: auction9-frontend/src/app/services/auction.service.ts [hunk 4]
```diff
@@ -91,4 +91,16 @@ export class AuctionService {
...
    }
+
+  /* postNewBid - create new bid for selected auction
+   * Method: POST
+   * Path: /auctions/id/new
+   */
+   createNewBid(auction, newBid) {
+     return this.http.post(`${environment.baseUrl}/auctions/${auction.auctionID}/new`,
+       {
+         auction,
```
Comment (mjovanovic) [inline auction9-frontend/src/app/services/auction.service.ts:102]: seems unnecessary?

File: auction9-backend/handler.js (hunk 1 above)
Comment (dangerousScript) [reply auction9-backend/handler.js:282]: Okay, other calls removed

File: auction9-backend/serverless.yml (hunk 2 above)
Comment (dangerousScript) [reply auction9-backend/serverless.yml:119]: Changed to 'bids'

File: auction9-backend/handler.js (hunk 3 above)
Comment (dangerousScript) [reply auction9-backend/handler.js:263]: Okay changed

File: auction9-frontend/src/app/services/auction.service.ts (hunk 4 above)
Comment (dangerousScript) [reply auction9-frontend/src/app/services/auction.service.ts:102]: remove auction from body as we will use auctionid from path parameters

**Review by mjovanovic — APPROVED** [review]

**Pull request merged**